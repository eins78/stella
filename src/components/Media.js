import React from 'react'
import {parse as parseUrl} from 'url'
import {extname} from 'path'

const Media = ({src, alt, caption}) => {
  if (!src) return false

  // TMP: only support images
  if (!['.jpg', '.jpeg', '.png', '.gif'].includes(extname(src))) {
    return false
  }

  // point to media folder if src is local path
  const imgSrc = parseUrl(src).host ? src : `/media/${src}`

  return (<div id='media'>
    <figure>
      <img alt={alt || caption} src={imgSrc} />
      {caption && <figcaption>{caption}</figcaption>}
    </figure>
  </div>)
}

export default Media
