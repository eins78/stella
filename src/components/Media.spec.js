/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import Media from './Media'

it('renders', () => {
  const component = renderer.create(
    <Media src='img.jpg' caption='caption txt' alt='alt txt' />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
