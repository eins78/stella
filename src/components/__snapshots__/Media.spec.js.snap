exports[`test renders 1`] = `
<div
  id="media">
  <figure>
    <img
      alt="alt txt"
      src="/media/img.jpg" />
    <figcaption>
      caption txt
    </figcaption>
  </figure>
</div>
`;
